### PR TITLE
docs: add button icon tooltip example

### PR DESCRIPTION
**Pull request by sissbruecker**
Extends the icon button example to show how to add a tooltip.

Also improve admonition to mention that the Flow `Button` component applies the `icon` theme automatically.

### CODE REVIEW

**Review by jouni — COMMENTED** [review]
Does the tooltip actually work as the label in this case, and not just as the description? I think those result in different screen reader behavior.

**Comment by sissbruecker** [comment]
Good point, the tooltip sets `aria-describedby`, so I guess those are not equivalent? The current wording kind of makes it sound like they are. Maybe it should be rephrased to indicate that a tooltip can be additionally added, but an `aria-label` is always recommended.

File: frontend/demo/component/button/button-icons.ts [hunk 1]
```diff
@@ -21,12 +22,18 @@ export class Example extends LitElement {
...
+        <!-- Icon button using an aria-label to provide textual alternative 
+             to screen readers -->
         <vaadin-button theme="icon" aria-label="Add item">
           <vaadin-icon icon="vaadin:plus"></vaadin-icon>
         </vaadin-button>
 
-        <vaadin-button theme="icon" aria-label="Close">
+        <!-- Icon button using a tooltip to provide textual alternative 
+             to screen readers, as well as when hovering the button with 
+             a pointing device, or focusing the button using keyboard -->
```
Comment (vursen) [inline frontend/demo/component/button/button-icons.ts:33]:
```suggestion
             a pointing device, or focusing the button using the keyboard -->
```

File: src/main/java/com/vaadin/demo/component/button/ButtonIcons.java [hunk 2]
```diff
@@ -13,13 +13,18 @@
...
         // tag::snippet[]
+        // Icon button using an aria-label to provide textual alternative
+        // to screen readers
         Button plusButton = new Button(new Icon(VaadinIcon.PLUS));
         plusButton.addThemeVariants(ButtonVariant.LUMO_ICON);
         plusButton.getElement().setAttribute("aria-label", "Add item");
 
+        // Icon button using a tooltip to provide textual alternative
+        // to screen readers, as well as when hovering the button with
+        // a pointing device, or focusing the button using keyboard
```
Comment (vursen) [inline src/main/java/com/vaadin/demo/component/button/ButtonIcons.java:24]:
```suggestion
        // a pointing device, or focusing the button using the keyboard
```

**Review by vursen — APPROVED** [review]

**Comment by sissbruecker** [comment]
I restored setting the `aria-label`, and changed the text so setting a tooltip is not interpreted as an alternative to setting an `aria-label`.

File: src/main/java/com/vaadin/demo/component/button/ButtonIcons.java [hunk 3]
```diff
@@ -13,13 +13,18 @@
 public class ButtonIcons extends Div {
     public ButtonIcons() {
         // tag::snippet[]
+        // Icon button using an aria-label to provide textual alternative
```
Comment (vursen) [inline src/main/java/com/vaadin/demo/component/button/ButtonIcons.java:16]:
```suggestion
        // Icon button using an aria-label to provide a textual alternative
```

**Review by vursen — APPROVED** [review]

File: src/main/java/com/vaadin/demo/component/button/ButtonIcons.java [hunk 4]
```diff
@@ -13,13 +13,18 @@
...
         Button plusButton = new Button(new Icon(VaadinIcon.PLUS));
         plusButton.addThemeVariants(ButtonVariant.LUMO_ICON);
         plusButton.getElement().setAttribute("aria-label", "Add item");
 
+        // Icon button using a tooltip to provide textual description
+        // of the action that it triggers
         Button closeButton = new Button(new Icon(VaadinIcon.CLOSE_SMALL));
+        plusButton.getElement().setAttribute("aria-label", "Close");
         closeButton.addThemeVariants(ButtonVariant.LUMO_ICON);
-        closeButton.getElement().setAttribute("aria-label", "Close");
```
Comment (jouni) [inline src/main/java/com/vaadin/demo/component/button/ButtonIcons.java:22]: Looks like these got mixed up. The `plusButton` now as the aria-label "Close", and the `closeButton` doesn't have an aria-label.

**Review by jouni — DISMISSED** [review]

File: src/main/java/com/vaadin/demo/component/button/ButtonIcons.java (hunk 4 above)
Comment (sissbruecker) [reply src/main/java/com/vaadin/demo/component/button/ButtonIcons.java]: Thanks, stupid copy+paste error. Fixed.

**Pull request merged**